module/User/src/User/Mapper/Session.php

```diff
@@ -63,6 +75,45 @@ public function persist($session)
     }
 
     /**
+     * Detach an entity from the entity manager.
+     *
+     * @param $entity
+     */
+    public function detach($entity)
+    {
+        $this->em->detach($entity);
+    }
+
+    /**
+     * Re-attach a user to the entity manager.
+     *
+     * @param entity $entity
+     *
+     * @return entity
+     */
+    public function merge($entity)
+    {
+        return $this->em->merge($entity);
```

Comment:
I know that merge has some side-effects to the database. In the sense that I thought it might push local changes back to the database in some cases. Maybe we should insure this shouldn't happen?